google/cloud/spanner_dbapi/connection.py

```diff
@@ -377,6 +387,22 @@ def close(self):
 
         self.is_closed = True
 
+    @check_not_closed
+    def begin(self):
+        """
+        Marks the transaction as started.
```

Comment:
See also my comment above: This should bring the connection into the same state as it would be if `autocommit=false` and it has not executed any statements yet. That means that the connection now has a transaction, but that transaction is not yet known to Cloud Spanner. That again means that it is possible to change the type of transaction (read-only / read/write), or other properties of the transaction.

E.g. in the future we might support more isolation levels than just `Serializable`. A statement like `set transaction isolation level repeatable read` is something that should be allowed after executing `BEGIN`, but not after having executed something like `SELECT * FROM foo` in that transaction.

Reply:
As discussed there is already an existing property inside_transaction(). Renamed the property